### PR TITLE
Fix xml include files

### PR DESCRIPTION
**Pull request by akva2**
"Small" bug introduced when we moved to tinyxml2. After include was injected a closing tag was added for the currently processed element. This meant we got an early closing tag which broke the structure of the XML document.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this with downstreams please

File: src/SIM/Test/with_include.xml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <simulation>
   <include>src/SIM/Test/included.xml</include>
+   <someothertag>is_here</someothertag>
```
Comment (kmokstad) [inline src/SIM/Test/with_include.xml:3]: inconsistent indent also here

File: src/SIM/XMLInputBase.C [hunk 2]
```diff
@@ -56,7 +56,7 @@ struct IncludeInjector : public tinyxml2::XMLVisitor
           return false;
       }
       currElem->InsertEndChild(inc_doc.RootElement()->DeepClone(&new_doc));
-      include_found = true;
+      include_found = include_processed = true;
     } else {
       e = new_doc.NewElement(elem.Name());
```
Comment (kmokstad) [inline src/SIM/XMLInputBase.C:61]: The temporary `e` is defined outside this block but only (afics) used inside it. Wouldn't that cause some static code check warnings?

**Review by kmokstad — APPROVED** [review]
Looks fine. Although I found it slightly confusing with the two bool variables plus the pointer to judge the status of the include tag. Maybe one int (or char) with value 0, 1 and 2 instead would be more clear? But that's up to you.

File: src/SIM/Test/with_include.xml (hunk 1 above)
Comment (akva2) [reply src/SIM/Test/with_include.xml]: fixed.

File: src/SIM/XMLInputBase.C (hunk 2 above)
Comment (akva2) [reply src/SIM/XMLInputBase.C]: possibly. reduced the scope.

**Comment by akva2** [comment]
jenkins build this with downstreams please

**Comment by akva2** [comment]
As one of the bools are user facing to I prefer to keep the two variables since the value convention would leak outside the class. I marked one of them private though.

**Pull request merged**